pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BinaryOperatorTransformFunction.java

```diff
@@ -422,6 +422,11 @@ private void fillLongResultArray(ValueBlock valueBlock, float[] leftValues, int
   }
 
   private void fillFloatResultArray(ValueBlock valueBlock, float[] leftValues, int length) {
+    // handling float double comparison, skipping the right CAST function and evaluate the result.
```

Comment:
This solution can only handle float-double comparison in the `SELECT` clause. Not able to handle similar comparisons in the `WHERE` clause.

I am looking for a better approach to handle both scenarios. 

@Jackie-Jiang @walterddr Do you think float-double comparison can occur in some other SQL clause?